Tests/SymbolKitTests/SymbolGraph/SymbolTests.swift

```diff
@@ -233,7 +233,113 @@ class SymbolTests: XCTestCase {
         let symbol = try JSONDecoder().decode(SymbolGraph.Symbol.self, from: inputGraph)
         XCTAssertNil(symbol.mixins[SymbolGraph.Symbol.Location.mixinKey])
     }
+ 
+    /// Check that an Extension mixin keeps the `typeKind` information if available and doesn't throw if it is absent.
+    func testOptionalExtensionTypeKindCoding() throws {
+        let inputGraphWithTypeKindInformation = """
+{
```

Comment:
Since the only part of the symbol graph we're testing here is the `swiftExtension` block, should we instead use just the following as our string:

```
{
     "extendedModule": "ExtendedModule",
     "typeKind": "class"
}
```

and decode using `JSONDecoder().decode(SymbolGraph.Symbol.Swift.Extension.self, from: …)`? That way, the test is more focused and easier to maintain.